### ISSUE TITLE
Filters break adding resources

### ISSUE THREAD

**Issue by max-allan-surevine** [awslabs/aws-perspective#46]
**Describe the bug**
White screen when filtering and adding more than a few resources. 

**To Reproduce**
1. Add a filter for AWS::CloudFormation::Stack and AWS::TAGS::TAG
2. Add more than a couple of resources to the screen
3. Watch the screen go white (entire Chrome window, Perspective toolbar, everything gone, like looking at about:blank ). 
4. Reload and you're back to an empty canvas (toolbar and grid return)

**Expected behavior**
Not to whitescreen but to show the resources, like it does when I disable the filters, I can add many.

**Browser **
Chrome Version 85.0.4183.121 (Official Build) (64-bit) on OSX

**Comment by thebutler12** [comment]
Hey, I wasn't able to replicate this following your instructions, but i was able to replicate something similar following these steps:

1. create an architecture diagram using a lambda function
2. open the filters menu and select AWS::TAGS::TAG and AWS::CloudFormation::Stack
3. Expand a resource in the the architecture diagram and it excludes the filtered resources
4. open the filters menu and deselect AWS::TAGS::TAG and AWS::CloudFormation::Stack
5. Expand a resource in the architecture diagram and then the UI fails and the error can be seen.

We will investigate this further and update this ticket.

**Comment by max-allan-surevine** [comment]
Thanks @thebutler12 . I can start with an empty canvas, enable 2 or more filters and start adding objects. A couple of EC2s, maybe an ALB, doesn't seem to matter what, but more than 3 or 4 objects will kill it for me.

From the console logs, it sounds like it is losing track of which objects are present and trying to create a edge to an object that a filter removed.

NB Logs from Chrome dev tools :

```
react-dom.production.min.js:209 Error: Can not create edge `2d888968f76dda13a16e5c9ea809e512-cad5c97d5ba44992eed0e3575b14434c` with nonexistant target `cad5c97d5ba44992eed0e3575b14434c`
    at le (cytoscape.cjs.js:811)
    at ni.ri.restore (cytoscape.cjs.js:12945)
    at new ni (cytoscape.cjs.js:12604)
    at Bi.add (cytoscape.cjs.js:13354)
    at l (react-cytoscape.js:1)
    at react-cytoscape.js:1
    at Bi.batch (cytoscape.cjs.js:14357)
    at t.patch (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
es @ react-dom.production.min.js:209
n.callback @ react-dom.production.min.js:226
fi @ react-dom.production.min.js:131
is @ react-dom.production.min.js:212
pu @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
fu @ react-dom.production.min.js:248
Zs @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
Go @ react-dom.production.min.js:123
$o @ react-dom.production.min.js:122
Ks @ react-dom.production.min.js:230
ga @ react-dom.production.min.js:163
(anonymous) @ TreeMenuResourceMenu.js:185
Promise.then (async)
C @ TreeMenuResourceMenu.js:180
le @ TreeView.js:375
s.forwardRef.s.createElement.value.selectNode @ TreeView.js:388
onClick @ TreeItem.js:186
s @ react-dom.production.min.js:14
p @ react-dom.production.min.js:14
(anonymous) @ react-dom.production.min.js:14
m @ react-dom.production.min.js:15
at @ react-dom.production.min.js:52
ot @ react-dom.production.min.js:51
st @ react-dom.production.min.js:52
pt @ react-dom.production.min.js:56
L @ react-dom.production.min.js:287
U @ react-dom.production.min.js:19
Jt @ react-dom.production.min.js:70
Qt @ react-dom.production.min.js:69
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
N @ react-dom.production.min.js:287
Yt @ react-dom.production.min.js:68
react-dom.production.min.js:209 Error: Can not create edge `2d888968f76dda13a16e5c9ea809e512-cad5c97d5ba44992eed0e3575b14434c` with nonexistant target `cad5c97d5ba44992eed0e3575b14434c`
    at le (cytoscape.cjs.js:811)
    at ni.ri.restore (cytoscape.cjs.js:12945)
    at new ni (cytoscape.cjs.js:12604)
    at Bi.add (cytoscape.cjs.js:13354)
    at l (react-cytoscape.js:1)
    at react-cytoscape.js:1
    at Bi.batch (cytoscape.cjs.js:14357)
    at t.patch (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
es @ react-dom.production.min.js:209
n.callback @ react-dom.production.min.js:226
fi @ react-dom.production.min.js:131
is @ react-dom.production.min.js:212
pu @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
fu @ react-dom.production.min.js:248
Zs @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
Go @ react-dom.production.min.js:123
$o @ react-dom.production.min.js:122
Ks @ react-dom.production.min.js:230
ga @ react-dom.production.min.js:163
(anonymous) @ TreeMenuResourceMenu.js:185
Promise.then (async)
C @ TreeMenuResourceMenu.js:180
le @ TreeView.js:375
s.forwardRef.s.createElement.value.selectNode @ TreeView.js:388
onClick @ TreeItem.js:186
s @ react-dom.production.min.js:14
p @ react-dom.production.min.js:14
(anonymous) @ react-dom.production.min.js:14
m @ react-dom.production.min.js:15
at @ react-dom.production.min.js:52
ot @ react-dom.production.min.js:51
st @ react-dom.production.min.js:52
pt @ react-dom.production.min.js:56
L @ react-dom.production.min.js:287
U @ react-dom.production.min.js:19
Jt @ react-dom.production.min.js:70
Qt @ react-dom.production.min.js:69
t.unstable_runWithPriority @ scheduler.production.min.js:19
Ho @ react-dom.production.min.js:122
N @ react-dom.production.min.js:287
Yt @ react-dom.production.min.js:68
cytoscape.cjs.js:811 Uncaught (in promise) Error: Can not create edge `2d888968f76dda13a16e5c9ea809e512-cad5c97d5ba44992eed0e3575b14434c` with nonexistant target `cad5c97d5ba44992eed0e3575b14434c`
    at le (cytoscape.cjs.js:811)
    at ni.ri.restore (cytoscape.cjs.js:12945)
    at new ni (cytoscape.cjs.js:12604)
    at Bi.add (cytoscape.cjs.js:13354)
    at l (react-cytoscape.js:1)
    at react-cytoscape.js:1
    at Bi.batch (cytoscape.cjs.js:14357)
    at t.patch (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
    at t.value (react-cytoscape.js:1)
```

**Comment by svozza** [comment]
This has been fixed in v1.1.0, which is available here: https://aws.amazon.com/solutions/implementations/aws-perspective/?did=sl_card&trk=sl_card.
